### PR TITLE
docking improvements

### PR DESCRIPTION
**Pull request by schmolly**
mostly relative_docking improvements
fix test outdirs
fix dock_and_equilibrate bug

### CODE REVIEW

File: docking/dock_and_equilibrate.py [hunk 1]
```diff
@@ -181,21 +181,23 @@ def dock_and_equilibrate(
...
                 str(step).zfill(len(str(insertion_steps))),
-                f"ins",
+                "ins",
             )
 
         if report.too_much_force(ctxt, lamb, host_box, combined_bps, u_impls):
             print("Not calculating work (too much force)")
             calc_work = False
-            break
+            continue
```
Comment (proteneer) [inline docking/dock_and_equilibrate.py:190]: Just checking that you intended to continue as opposed to break

Comment (schmolly) [reply docking/dock_and_equilibrate.py:190]: yep, that was the bug! there's one spot where it should break and a different spot where it should continue. The bug was that if one compound in an sdf failed, the rest of the compounds in the set wouldn't run at all

File: tests/test_docking.py [hunk 2]
```diff
@@ -35,18 +35,18 @@ def test_pose_dock(self):
...
+        self.assertTrue(Path(f"{outdir}/43/43_pd_1000_guest.sdf").exists())
 
-        for f in Path("test_pd_outdir").glob("*/*.*"):
+        for f in Path(outdir).glob("*/*.*"):
             Path.unlink(f)
-        for d in Path("test_pd_outdir").glob("*"):
+        for d in Path(outdir).glob("*"):
             d.rmdir()
-        Path("test_pd_outdir").rmdir()
+        Path(outdir).rmdir()
```
Comment (proteneer) [inline tests/test_docking.py:49]: if the assertion/tests fail do the directories still get cleaned up? not sure if you need a context manager or a use a context managed tempfile/tempdir?

Comment (schmolly) [reply tests/test_docking.py:49]: good question! I'll have to google that

File: docking/rigorous_work.py [hunk 3]
```diff
@@ -21,28 +21,34 @@
 DELETION_MAX_LAMBDA = 1.0
 MIN_LAMBDA = 0.0
 INSERTION_STEPS = 501
-DELETION_STEPS = 501
 EQ1_STEPS = 5001
```
Comment (maxentile) [inline docking/rigorous_work.py:24]: Would it be helpful to expose any other of these variables as CLI arguments? (Maybe `INSERTION_STEPS` and `EQ1_STEPS`?)

File: docking/rigorous_work.py [hunk 4]
```diff
@@ -424,12 +445,20 @@ def main():
     parser.add_argument(
         "--no_outfiles", action="store_true", help="write no output pdb/sdf files"
     )
+    parser.add_argument(
+        "--deletion_steps", type=int, default=501, help="how many steps to delete over"
+    )
+    parser.add_argument(
+        "--num_deletions", type=int, default=10, help="number of deletions to do"
+    )
```
Comment (maxentile) [inline docking/rigorous_work.py:453]: Suggest to swap order of these to match the Python signature (although not technically required!)

File: docking/relative_docking.py [hunk 5]
```diff
@@ -18,24 +27,19 @@
 
 from docking import report
 
+
```
Comment (maxentile) [inline docking/relative_docking.py:30]: Can remove testsystems import on L26

**Review by maxentile — APPROVED** [review]
Nice work! These all look like clear improvements.

Replacing the `uuid`-named directory in tests with a `tempfile`-managed temporary directory, as @proteneer suggested, would probably be even a bit safer.

File: docking/rigorous_work.py (hunk 3 above)
Comment (schmolly) [reply docking/rigorous_work.py:24]: might as well!

**Review by proteneer — APPROVED** [review]
@schmolly  did you want to merge this? I don't remember where we left off on this...

**Comment by schmolly** [comment]
yes, I'm ready to merge! :)

**Comment by proteneer** [comment]
Great - feel free to merge once tests pass!

**Pull request merged**